### PR TITLE
Add basic format and query function

### PR DESCRIPTION
**Pull request by neithernut**
The idea behind this PR is to replace the current format and bbp functions in the buffer by a new pointer datatype which represents a format and allows easier conversion.
Blocks #602.
Note that this is still WIP as there are a few problems which are not solved yet.

### CODE REVIEW

File: src/util/egl.c [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#include "util/egl.h"
+
+static struct ws_egl_fmt_mapping const mappings[] = {
+    {
+        .shm_fmt = WL_SHM_FORMAT_ARGB8888,
+        .egl = { .fmt = 0, .type = 0},
+        .bbp = 8
+    },
+    {
+        .shm_fmt = WL_SHM_FORMAT_ARGB8888,
```
Comment (TheNeikos) [inline src/util/egl.c]: Didn't you mean using XRGB at this point?

Comment (neithernut) [inline src/util/egl.c]: Yep.

**Comment by neithernut** [comment]
I think I'm done here.
Note that the glew dependency is not satisfied by the CI. I'll push another PR with an appropriate addition to the build env (CMake).

---

Disregard that. It turned out that I didn't need to write a module for finding GLEW as CMake already provided one. @TheNeikos, please update the CI and trigger a rebuild.

---

I'm backing off once more, but still needing the GLES2-Headers installed on the CI.

**Comment by neithernut** [comment]
This PR should now finally be complete, but there are two things to note here:
- The CI currently lacks the GLES2 development files (as provided by mesa)
- As we _should_ only accept RGBA now, this _temporarily_ breaks the display functionality (as the DRM doesn't get what it wants). This will be fixed once #533 is merged (e.g. when we actually use EGL for compositing).

File: src/compositor/buffer/buffer.c [hunk 2]
```diff
@@ -217,13 +200,19 @@ ws_buffer_blit(
         return;
     }
 
+    struct ws_egl_fmt const* dest_fmt = ws_buffer_format(dest);
+    struct ws_egl_fmt const* src_fmt  = ws_buffer_format(src);
+    if (!(dest_fmt && src_fmt )) {
```
Comment (matthiasbeyer) [inline src/compositor/buffer/buffer.c]: Whitespace before `))`.

File: src/util/egl.c [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+static struct ws_egl_fmt const mappings[] = {
+    {
+        .shm_fmt = WL_SHM_FORMAT_RGBA8888,
+        .egl = { .fmt = GL_RGBA8_OES, .type = GL_UNSIGNED_INT},
+        .bbp = 4
+    },
+    {
+        .shm_fmt = WL_SHM_FORMAT_RGBX8888,
+        .egl = { .fmt = GL_RGBA8_OES, .type = GL_UNSIGNED_INT},
```
Comment (matthiasbeyer) [inline src/util/egl.c]: Missing whitespace before closing `}`, in line 45 as well...

**Comment by matthiasbeyer** [comment]
For me this looks sane, please fix the style annotations. Hopefully we can get this in as soon as the CI is prepared...

**Comment by TheNeikos** [comment]
the GLES2 headers are now in the image, please rebase to see that actual error message.

**Comment by neithernut** [comment]
Now this is embarrasing. I actually copy-pasted a typo I made when porting the build env to the modular approach. That typo somehow didn't cause the compilation to fail originially _or_ on my machine, so I didn't notice it until now.
Anyways, fixed.

**Comment by matthiasbeyer** [comment]
ACK.

**Pull request merged**